CADETProcess/optimization/optimizationProblem.py

```diff
@@ -2759,7 +2761,11 @@ def cached_steps(self):
     def cache_directory(self):
         """pathlib.Path: Path for results cache database."""
         if self._cache_directory is None:
-            _cache_directory = settings.working_directory / f'diskcache_{self.name}'
+            if 'XDG_CACHE_HOME' in os.environ:
+                start_time = int(datetime.now().timestamp() * 1000)
+                _cache_directory = Path(os.environ['XDG_CACHE_HOME']) / 'CADET-Process' / f'{self.name}_{start_time}'
+            else:
+                _cache_directory = settings.working_directory / f'diskcache_{self.name}'
```

Comment:
Should we keep the cache directory name consistent regardless whether `XDG_CACHE_HOME` is specified or not? I.e.  always use `f'{self.name}_{start_time}'`?

Reply:
After our discussion online: let's use the following path:
```
Path(os.environ['XDG_CACHE_HOME']) / 'CADET-Process' / hash(settings.working_directory) / f'diskcache_{self.name}')
```
This has the same limitatations as the regular cache dir (e.g. don't run two optimizations with the same name in the same working directory) and does not pollute cache space by creating too many directories that never the overwritten.